Source/BestProjectEver/Private/AI/BPE_PatrolPath.cpp

```diff
@@ -0,0 +1,36 @@
+// Fill out your copyright notice in the Description page of Project Settings.
+
+
+#include "AI/BPE_PatrolPath.h"
+
+#include "Components/SplineComponent.h"
+
+// Sets default values
+ABPE_PathFollowing::ABPE_PathFollowing()
+{
+	PrimaryActorTick.bCanEverTick = false;
+
+	SplineComponent = CreateDefaultSubobject<USplineComponent>(TEXT("SplineCompononent"));
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+FVector ABPE_PathFollowing::GetWorldLocationByIndex(const int32 Index) const
+{
+	if(IsValid(SplineComponent) && Index >= 0 && Index <= GetLastIndex())
+	{
+		return SplineComponent->GetLocationAtSplinePoint(Index,ESplineCoordinateSpace::World);
+	}
+	return GetActorLocation();
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+int32 ABPE_PathFollowing::GetLastIndex() const
+{
+	if(IsValid(SplineComponent))
+	{
+		return SplineComponent->GetNumberOfSplinePoints();
+	}
+	return 0;
```

Comment:
You could return INDEX_NONE here, considering that 0 is a valid index.

Reply:
![image](https://user-images.githubusercontent.com/85327912/173114208-b4be86ae-7fad-444d-a612-3eb31b300e72.png)